File: Programming And Websites/Python/Python Programs/insertionsort.py

def insertionSort(arr):

    if (n := len(arr)) <= 1:
        return
    for i in range(1, n):

        key = arr[i]
        j = i - 1

        while j >= 0 and key < arr[j]:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key


arr = [12, 11, 13, 5, 6]
print("Array is:")
print(arr)

insertionSort(arr)

print("After Sorting:")
print(arr)

File: Programming And Websites/DSA in C/Sorting/InsertionSort.c
#include <stdio.h>

int main()
{
    int arr[] = {4, 8, 2, 6, 0, 1, 5};

    printf("Array is: ");
    for (int i = 0; i < 7; i++)
    {
        printf("%d ", arr[i]);
    }

    // Insertion Sorting

    for (int i = 1; i < 7; i++)
    {
        for (int j = i; j > 0; j--)
        {
            if (arr[j - 1] > arr[j])
            {
                int temp = arr[j - 1];
                arr[j - 1] = arr[j];
                arr[j] = temp;
            }
        }
    }

    printf("\nSorted Array is: ");
    for (int i = 0; i < 7; i++)
    {
        printf("%d ", arr[i]);
    }

    return 0;
}